googlemock/include/gmock/gmock-actions.h

```diff
@@ -717,13 +713,9 @@ class ReturnAction {
    private:
     bool performed_;
     const std::shared_ptr<R> wrapper_;
-
-    GTEST_DISALLOW_ASSIGN_(Impl);
```

Comment:
Since the original intent is to make `Impl` not copy assignable. We should explicitly write
`Impl& operator=(const Impl&) = delete;`
not relying on the `const` member to disable the assignment operator.
You are right that `const` member makes move ctor call copy ctor even we declare move ctor `=default`, but I think it's better to make the intention clear if we want this type to be move constructible (even though under the hood it invokes copy constructor). If the `const` somehow is removed later, we will automatically get the optimization.
However, feel free to NOT declare the move ctor here. It's technically a separate thing that could be handled separately.

Reply:
If we `=delete` the copy assignment operator, then (to avoid relying on deprecated behavior) we must also put something for the copy constructor (e.g. `=default` it). Clang doesn't actually warn on this specific deprecated behavior yet, but I just filed https://bugs.llvm.org/show_bug.cgi?id=45634 about it.

Anyway, I contend that in all these places where I removed `GTEST_DISALLOW_ASSIGN_ `, there _was_ no discernable intent. There's no obvious reason that these types _should_ be non-assignable; and on the flip side, most of them are accidentally non-assignable anyway even if you remove the deleted assignment operator. I think it's better to follow the Rule of Zero unless we have a specific metaprogramming reason that we _need_ `std::is_assignable<T&, const T&>::value` to be `false` or else code would break. (And in that case, there should be unit tests to regression-test that code that would have broken.)